vcf/gVcf.go

```diff
@@ -192,63 +191,47 @@ func PhasedToString(phased bool) string {
 	}
 }
 
+//ReorderSampleColumns reorganizes the Samples slice based on a samples []int16 specification list.
 func ReorderSampleColumns(input *Vcf, samples []int16) *Vcf {
-	columnData := make([]string, 0, len(samples))
-	words := strings.Split(input.Notes, "\t")
-	var i int
-	for i = 0; i < len(samples); i++ {
-		columnData = append(columnData, words[samples[i]])
+	outSamples := make([]GenomeSample, 0, len(samples))
+	for i := 0; i < len(samples); i++ {
+		outSamples = append(outSamples, input.Samples[samples[i]])
 	}
-	input.Notes = strings.Join(columnData, "\t")
+	input.Samples = outSamples
 	return input
 }
 
 func PrintReOrder(v *Vcf, samples []int16) {
-	Genotypes := VcfToGvcf(ReorderSampleColumns(v, samples))
-	log.Printf("%s\t%d\t%s\t%s\t%s\n", v.Chr, v.Pos, v.Ref, v.Alt, GenotypeToString(Genotypes))
-}
-
-func GenotypeToString(sample *GVcf) string {
-	var answer string = ""
-	for i := 0; i < len(sample.Genotypes); i++ {
-		answer += helperGenotypeToString(sample, i)
-	}
-	return answer
-}
-
-func helperGenotypeToString(sample *GVcf, i int) string {
-	if sample.Genotypes[i].AlleleOne < 0 {
-		return "NoData "
-	} else {
-		if i == len(sample.Genotypes)-1 {
-			return fmt.Sprintf("%d%s%d=%s%s%s", sample.Genotypes[i].AlleleOne, PhasedToString(sample.Genotypes[i].Phased), sample.Genotypes[i].AlleleTwo, dna.BasesToString(sample.Seq[sample.Genotypes[i].AlleleOne]), PhasedToString(sample.Genotypes[i].Phased), dna.BasesToString(sample.Seq[sample.Genotypes[i].AlleleTwo]))
-		} else {
-			return fmt.Sprintf("%d%s%d=%s%s%s\t", sample.Genotypes[i].AlleleOne, PhasedToString(sample.Genotypes[i].Phased), sample.Genotypes[i].AlleleTwo, dna.BasesToString(sample.Seq[sample.Genotypes[i].AlleleOne]), PhasedToString(sample.Genotypes[i].Phased), dna.BasesToString(sample.Seq[sample.Genotypes[i].AlleleTwo]))
-		}
-	}
+	vReorder := ReorderSampleColumns(v, samples)
+	log.Printf("%s\t%d\t%s\t%s\t%s\n", v.Chr, v.Pos, v.Ref, v.Alt, SamplesToString(vReorder.Samples))
 }
 
-func GenotypeToStringNew(sample []GenomeSample) string {
+func SamplesToString(sample []GenomeSample) string {
 	var answer string = ""
 	for i := 0; i < len(sample); i++ {
-		answer += helperGenotypeToStringNew(sample, i)
+		answer += HelperSamplesToString(sample, i)
 	}
 	return answer
 }
 
 //helperGenotypeToStringNew uses just an array of GenomeSample structs to write to a string for simple gVCFs with just the allele info in notes.
-func helperGenotypeToStringNew(sample []GenomeSample, i int) string {
+func HelperSamplesToString(sample []GenomeSample, i int) string {
+	var answer string
 	if sample[i].AlleleOne < 0 {
-		return "noData "
+		answer = "noData "
```

Comment:
No space at end?

Reply:
Eric set this up with the space, not sure if it has significance. I'll remove.